framework/auth/views.py

```diff
@@ -251,6 +279,88 @@ def confirm_email_get(token, auth=None, **kwargs):
     ))
 
 
+@collect_auth
+def verified_email_get(auth=None, **kwargs):
+    """Called at login to see if there are emails to add or users to merge.
+    methods: GET
+    """
+    # user = User.load(kwargs['uid'])
+    user = auth.user
+    verified_emails = []
+    for token in user.email_verifications:
+        if user.verify_token(token):
+            if user.email_verifications[token]['confirmed']:
+                try:
+                    user_merge = User.find_one(Q('emails', 'eq', user.email_verifications[token]['email'].lower()))
+                except NoResultsFound:
+                    user_merge = False
+
+                verified_emails.append({'address': user.email_verifications[token]['email'],
+                                        'token': token,
+                                        'confirmed': user.email_verifications[token]['confirmed'],
+                                        'user_merge': user_merge.email if user_merge else False})
+    return verified_emails
+
+
+@collect_auth
+def verified_email_remove(auth=None, **kwargs):
+    """Called at login if user cancels their merge or email add.
+    methods: PUT
+    """
+    user = auth.user
+    confirmed_email = request.json.get('address')
```

Comment:
A 400 error should be returned if this isn't provided.

Also, `request.json` is [deprecated](http://flask.pocoo.org/docs/0.10/api/#flask.Request.json). Use `.get_json()` instead.